services/appflowy-collaborate/src/group/cmd.rs

```diff
@@ -155,7 +155,11 @@ where
         let origin = first_message.origin().clone();
         let msg_id = first_message.msg_id();
         let object_id = first_message.object_id().to_string();
-        let ack = CollabAck::new(origin, object_id, msg_id, 0);
+
+        // when the group with given id is not found and the the first message is not init sync.
+        // Return AckCode::CannotApplyUpdate to the client and then client will send the init sync message.
+        let ack =
+          CollabAck::new(origin, object_id, msg_id, 0).with_code(AckCode::CannotApplyUpdate);
```

Comment:
Cannot we just close connection here and force client to reconnect?:

1. When connecting, client will send init sync message anyway.
2. There may be some expectations about what state the client is in when doing init sync with regards to things like sequence numbers etc. Resetting the connection makes sure that we start at init state.

Reply:
close websocket connect will force all collab objects to start init sync. But here I just force the corresponding collab to start init sync. For old version appflowy client, we can stick to this approach. For new version, we introduce new sync protocol

Reply:
What's the original reason behind the update that cannot be applied?